starlette_context/plugins/base.py

```diff
@@ -52,18 +52,20 @@ def __init__(
         force_new_uuid: bool = False,
         version: int = 4,
         validate: bool = True,
+        error_response: Optional[Response] = None,
```

Comment:
I think this callback func mentioned above could create a response based on the error, wdyt?

Reply:
It seems this is the approach taken by the the ASGI RateLimit middleware 
https://github.com/abersheeran/asgi-ratelimit/blob/master/ratelimit/core.py
taking the callback from init (I think that makes it a signature similar to `def callback(exc: Exception) -> Response`):
```
        on_auth_error: Optional[Callable[[Exception], Awaitable[ASGIApp]]] = None,
```
and using it on error cases
```
                    except Exception as exc:
                        if self.on_auth_error is not None:
                            reponse = await self.on_auth_error(exc)
                            return await reponse(scope, receive, send)
                        raise exc
```
I guess that might work, although completely shifts the burden on users to define the callback and only gives the exception.
With a custom exception, that might be acceptable...